src/main/java/com/fakng/fakngagrgtr/parser/amazon/AmazonParser.java

```diff
@@ -0,0 +1,106 @@
+package com.fakng.fakngagrgtr.parser.amazon;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+
+import com.fakng.fakngagrgtr.parser.ApiParser;
+import com.fakng.fakngagrgtr.parser.LocationProcessor;
+import com.fakng.fakngagrgtr.persistent.company.CompanyRepository;
+import com.fakng.fakngagrgtr.persistent.location.Location;
+import com.fakng.fakngagrgtr.persistent.vacancy.Vacancy;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClient.ResponseSpec;
+
+@Component
+public class AmazonParser extends ApiParser {
+
+    private static final DateTimeFormatter DATE_TIME_FORMATTER =
+            DateTimeFormatter.ofPattern("MMMM d, u", Locale.ENGLISH);
+
+    public AmazonParser(WebClient webClient,
+            CompanyRepository companyRepository,
+            LocationProcessor locationProcessor,
+            @Value("${url.amazon}") String url) {
+        super(webClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected String getCompanyName() {
+        return "Amazon";
+    }
+
+    @Override
+    protected List<Vacancy> getAllVacancies() {
+        int offset = 0;
+        ResponseDTO firstPage = getPage(offset);
+        List<Vacancy> vacancies = new ArrayList<>(processPageResponse(firstPage));
+        while (vacancies.size() < firstPage.getHits() && vacancies.size() < 10000) {
+            ResponseDTO page = getPage(vacancies.size());
+            vacancies.addAll(processPageResponse(page));
+        }
+        return vacancies;
+    }
+
+    private List<Vacancy> processPageResponse(ResponseDTO response) {
+        return Optional.of(response.getJobs())
+                .orElse(Collections.emptyList())
+                .stream()
+                .map(this::createVacancy)
+                .toList();
+    }
+
+    private Vacancy createVacancy(VacancyDTO dto) {
+        Vacancy vacancy = new Vacancy();
+//        vacancy.setId(dto.getId()); // need to create string field for containing native id
```

Comment:
Yep, that's right
But we don't need commented code here

This will be job_id field. There's a task available on board for creation of this one

Reply:
Removed this.